### PR TITLE
Adds new component Menu

### PR DESCRIPTION
**Pull request by tumentumurchudur**
This PR adds a new component `Menu`.

## Props
- alignItems: string - Allows users to align menu items either to the right or left
- isOpen: bool - determines whether the menu items are open
- items: array - defines the menu items
- onClick: func - callback when a user clicks the Menu component
- primaryColor: string - determines the hover color for the menu items

ezgif com-video-to-gif 5

### CODE REVIEW

File: src/components/Menu.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+const Icon = require('../components/Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const Menu = React.createClass({
+  propTypes: {
+    alignItems: React.PropTypes.oneOf(['left', 'right']),
+    isOpen: React.PropTypes.bool,
+    items: React.PropTypes.arrayOf(React.PropTypes.shape({
```
Comment (jtanner) [inline src/components/Menu.js]: Nice use of `React.PropTypes.shape` 👍

Comment (jtanner) [inline src/components/Menu.js]: Nice use of `React.PropTypes.shape` 👍

**Comment by tumentumurchudur** [comment]
Conflicts with master have been resolved.

**Comment by jtanner** [comment]
@tumentumurchudur `demo/app.js` shouldn't have been added back in, it was removed by @phantomxc in #432 so we don't have to maintain two places for demos/docs. Instead of `demo/app.js` you should add some documentation and a demo to `docs/`.

**Comment by tumentumurchudur** [comment]
Yes, you are right. It is removed now.

**Comment by jtanner** [comment]
@tumentumurchudur will you add docs to this too?

**Comment by tumentumurchudur** [comment]
Yes, I'm working on it. 👍

**Comment by tumentumurchudur** [comment]
Adds documentation for the `Menu` component.

screen shot 2016-09-23 at 10 00 45 am

**Comment by jtanner** [comment]
:shipit:

**Pull request merged**